app/models/subscription_trial.rb

```diff
@@ -0,0 +1,51 @@
+class SubscriptionTrial < ApplicationRecord
+  SUBSCRIPTION_LENGTH = 14.days
+
+  belongs_to :user
+
+  validates :user_id, presence: true, uniqueness: true
+
+  def ended_at
+    created_at + SUBSCRIPTION_LENGTH
```

Comment:
Since this is going to be completely static maybe we should store this in the DB instead of calculating it every time?

Reply:
I disagree here actually, for the same reason we calculate (rather than store) `Segment#realtime_end_ms`/`Segment#gametime_end_ms` -- it guarantees data integrity because we only have one source of data truth. We can never have a segment end that doesn't make sense, and if we do anyway because of a logic bug, when we fix it we are not left over with a bunch of incorrect data we need to fix in a migration. We are trading some computation time for some data guarantees.

In the case of trials it's an immeasurably small computation and a small-ish data guarantee, but I think the principle still holds.

I can see two advantages of storing this in the database separately:

1. if we decide to later change the subsciption trial length, it would only change for new trials, which might be preferable depending on the change.
2. if we have a special situation where an admin wants to extend someone's trial (or perhaps make it indefinite), they can edit `ended_at` which "makes sense", rather than edit `created_at` which, while it would work, wouldn't really "make sense", or they would have to add a new feature grantor for an admin-granted free subscription or somesuch.

But IMO they are not big enough advantages.

Reply:
Makes perfect sense when I think about it that way, thanks for the explanation!